### PR TITLE
Replace has_one with attribute in template

### PR DESCRIPTION
**Pull request by bf4**
has_one is no longer supported

this way we generate valid templates

### CODE REVIEW

File: lib/generators/serializer/templates/serializer.rb [hunk 1]
```diff
@@ -3,6 +3,6 @@ class <%= class_name %>Serializer < <%= parent_class_name %>
   attributes <%= attributes_names.map(&:inspect).join(", ") %>
 end
 <% association_names.each do |attribute| -%>
-  has_one :<%= attribute %>
+  attribute :<%= attribute %>
```
Comment (kurko) [inline lib/generators/serializer/templates/serializer.rb]: I believe it should be `belongs_to`, not `attribute`.

Comment (bf4) [inline lib/generators/serializer/templates/serializer.rb]: If you say so.  There's basically no usage tests around this code.

The generator defines [association_names](https://github.com/rails-api/active_model_serializers/blob/a8243761fedbff9877a957fe9b5f21e644bb8fee/lib/generators/serializer/serializer_generator.rb#L21) as

```ruby
def association_names
  attributes.select { |attr| attr.reference? }.map { |a| a.name.to_sym }
end
```

[belongs_to](https://github.com/rails-api/active_model_serializers/blob/a8243761fedbff9877a957fe9b5f21e644bb8fee/lib/active_model/serializer.rb#L24)

```ruby
    # Defines an association in the object that should be rendered.
    #
    # The serializer object should implement the association name
    # as a method which should return an object when invoked. If a method
    # with the association name does not exist, the association name is
    # dispatched to the serialized object.
    def self.belongs_to(*attrs)
      associate(:belongs_to, attrs)
    end
```

Whereas attribute

```ruby
    def self.attribute(attr, options = {})
      key = options.fetch(:key, attr)
      @_attributes.concat [key]
      define_method key do
        object.read_attribute_for_serialization(attr)
      end unless method_defined?(key)
    end
```

I'm not really sure where `references?` comes from. Seems to be a schema definition for a `has_many`, no?

**Comment by kurko** [comment]
Nevermind. I misread this PR before.

**Comment by bf4** [comment]
Thanks!

**Comment by bf4** [comment]
@kurko Any desire to revert this since was re-added in https://github.com/rails-api/active_model_serializers/pull/725 ? I'm happy to submit a PR.

**Comment by kurko** [comment]
Yes, definitely.

**Comment by bf4** [comment]
Done

**Pull request merged**